mmdet/integration/nncf/compression.py

```diff
@@ -111,9 +116,60 @@ def get_inputs(self, dataloader_output):
     nncf_config = NNCFConfig(cfg.nncf_config)
     logger = get_root_logger(cfg.log_level)
 
-    if data_loader_for_init:
-        wrapped_loader = MMInitializeDataLoader(data_loader_for_init)
-        nncf_config = register_default_init_args(nncf_config, wrapped_loader, device=next(model.parameters()).device)
+    def model_eval_fn(model):
+        """
+        Runs evaluation of the model on the validation set and
+        returns the target metric value.
+        Used to evaluate the original model before compression
+        if NNCF-based accuracy-aware training is used.
+        """
+        if val_dataloader is None:
+            raise RuntimeError('Cannot perform model evaluation on the validation '
+                               'dataset since the validation data loader was not passed '
+                               'to wrap_nncf_model')
+        from mmdet.apis import single_gpu_test, multi_gpu_test
+        from functools import partial
+
+        metric_name = nncf_config.get('target_metric_name', 'bbox_mAP')
+        forward_backup = model.forward
+        model_forward = type(model).forward
+        model.forward = model_forward.__get__(model)
+        model.forward = partial(model.forward, return_loss=False)
+        prepared_model = prepare_mmdet_model_for_execution(model, cfg, distributed)
+
+        logger.info(f'Calculating an original model accuracy')
+
+        if distributed:
+            dist_eval_res = [None]
+            results = multi_gpu_test(prepared_model, val_dataloader, gpu_collect=True)
+            if torch.distributed.get_rank() == 0:
+                eval_res = val_dataloader.dataset.evaluate(results)
+                if metric_name not in eval_res:
+                    raise RuntimeError(f'Cannot find {metric_name} metric in '
+                                       'the evaluation result dict')
+                dist_eval_res[0] = eval_res
+
+            torch.distributed.broadcast_object_list(dist_eval_res, src=0)
+            model.forward = forward_backup
+            return dist_eval_res[0][metric_name]
+        else:
+            results = single_gpu_test(prepared_model, val_dataloader, show=False)
+            eval_res = val_dataloader.dataset.evaluate(results)
+
+            if metric_name not in eval_res:
+                raise RuntimeError(f'Cannot find {metric_name} metric in '
+                                   'the evaluation result dict')
+
+            model.forward = forward_backup
+            return eval_res[metric_name]
+
+    wrapped_loader = None
+    if dataloader_for_init:
+        wrapped_loader = MMInitializeDataLoader(dataloader_for_init)
+    eval_fn = model_eval_fn if is_accuracy_aware else None
+    nncf_config = register_default_init_args(nncf_config, wrapped_loader,
```

Comment:
When `wrapped_loader`  is None `register_default_init_args` fails with 
```
nncf/nncf/torch/initialization.py", line 74, in wrap_dataloader_for_init
    loaded_item = next(iter(data_loader))
TypeError: 'NoneType' object is not iterable
```
For example, this happens in `tools/test.py`

Reply:
also `wrap_nncf_model` call in `tools/test.py` should be updated

Reply:
Thank you for the comment! I will fix it

Reply:
Done